polygeist/tools/cgeist/Test/Verification/ptrarith.c

```diff
@@ -0,0 +1,76 @@
+// RUN: cgeist %s --function=* -S | FileCheck %s
+
+#include <stddef.h>
+
+// CHECK-LABEL:   func.func @f0(
+// CHECK-SAME:                  %[[VAL_0:.*]]: memref<?xi32>,
+// CHECK-SAME:                  %[[VAL_1:.*]]: i32) -> memref<?xi32>
+// CHECK:           %[[INDEX:.*]] = arith.index_cast %[[VAL_1]] : i32 to index
+// CHECK:           %[[ADD:.*]] = "polygeist.subindex"(%[[VAL_0]], %[[INDEX]]) : (memref<?xi32>, index) -> memref<?xi32>
+// CHECK:           return %[[ADD]] : memref<?xi32>
+// CHECK:         }
+
+int *f0(int *ptr, int index) {
+  return ptr + index;
+}
+
+// CHECK-LABEL:   func.func @f1(
+// CHECK-SAME:                  %[[VAL_0:.*]]: i64,
+// CHECK-SAME:                  %[[VAL_1:.*]]: memref<?xi32>) -> memref<?xi32>
+// CHECK:           %[[INDEX:.*]] = arith.index_castui %[[VAL_0]] : i64 to index
```

Comment:
Now we generate `index_castui` for unsigned.

Reply:
Should we use the same arith instruction for the test case at line 13

Reply:
Whitney is right:
```
index_cast(signed) -> index
index_castui(unsigned) -> index
```